core/src/main/scala/kafka/utils/ToolsUtils.scala

```diff
@@ -64,4 +65,18 @@ object ToolsUtils {
         println(s"%-${maxLengthOfDisplayName}s : $specifier".format(metricName, value))
     }
   }
+
+  /**
+   * This is a simple wrapper around `CommandLineUtils.printUsageAndDie`.
+   * It is needed for tools migration (KAFKA-14525), as there is no Java equivalent for return type `Nothing`.
+   * Can be removed once [[kafka.admin.ConsumerGroupCommand]], [[kafka.tools.ConsoleConsumer]]
+   * and [[kafka.tools.ConsoleProducer]] are migrated.
+   *
+   * @param parser Command line options parser.
+   * @param message Error message.
+   */
+  def printUsageAndDie(parser: OptionParser, message: String): Nothing = {
```

Comment:
I am thinking if we can add another parameter to this to say if we want to do an Exit(1) from this method or not. I see some places where Exit(1) is not being invoked from the scala layer.

Reply:
In the original code we always call exit, as the method name implies, so I think it's fine.